pandas/core/frame.py

```diff
@@ -6476,6 +6476,8 @@ def diff(self, periods=1, axis=0) -> "DataFrame":
         -----
         For boolean dtypes, this uses :meth:`operator.xor` rather than
         :meth:`operator.sub`.
+        The result is calculated according to current dtype in DataFrame,
+        however dtype of the result is always float64.
```

Comment:
My understanding from the issue, is that the main problem is the overflow, not the type.

I think adding this example from the issue to the examples, with a comment explaining why the results are counter-intuitive (the value and the type), could probably be more useful:
```python
>>> import numpy as np
>>> import pandas as pd
>>> 
>>> s = pd.Series([1, 0], dtype=np.uint8)
>>> s.diff()
0      NaN
1    255.0
dtype: float64
```

Reply:
done